src/rank_llm/retrieve_and_rerank.py

```diff
@@ -64,62 +83,55 @@ def retrieve_and_rerank(
+                    k=top_k_retrieve, 
+                    host=host
+                )
+            ]
+        else:
+            requests = Retriever.from_dataset_with_prebuilt_index(
+                dataset_name=dataset, retrieval_method=retrieval_method
+            )
+
     elif retrieval_mode == RetrievalMode.CUSTOM:
         requests = Retriever.from_custom_index(
             index_path=index_path, topics_path=topics_path, index_type=index_type
         )
     else:
         raise ValueError(f"Invalid retrieval mode: {retrieval_mode}")
-    print("Reranking:")
+    print(f"Retrieval complete!")
+    
+    # Reranking
+    print(f"Reranking and returning {top_k_rerank} passages...")
     reranker = Reranker(agent)
     for pass_ct in range(num_passes):
         print(f"Pass {pass_ct + 1} of {num_passes}:")
-        rerank_results = reranker.rerank_batach(
+        rerank_results = reranker.rerank_batch(
             requests,
-            rank_end=top_k_candidates,
-            window_size=min(window_size, top_k_candidates),
+            rank_end=top_k_retrieve,
+            window_size=min(window_size, top_k_retrieve),
             shuffle_candidates=shuffle_candidates,
             logging=print_prompts_responses,
             step=step_size,
+            populate_exec_summary=populate_exec_summary
         )
 
-        # generate trec_eval file & evaluate for named datasets only
-        if isinstance(dataset, str):
-            file_name = reranker.write_rerank_results(
-                retrieval_method.name,
-                rerank_results,
-                shuffle_candidates,
-                top_k_candidates=top_k_candidates,
-                pass_ct=None if num_passes == 1 else pass_ct,
-                window_size=window_size,
-                dataset_name=dataset,
-            )
-            if (
-                dataset in TOPICS
-                and dataset not in ["dl22", "dl22-passage", "news"]
-                and TOPICS[dataset] not in ["dl22", "dl22-passage", "news"]
-            ):
-                print("Evaluating:")
-                EvalFunction.eval(
-                    ["-c", "-m", "ndcg_cut.1", TOPICS[dataset], file_name]
-                )
-                EvalFunction.eval(
-                    ["-c", "-m", "ndcg_cut.5", TOPICS[dataset], file_name]
-                )
-                EvalFunction.eval(
-                    ["-c", "-m", "ndcg_cut.10", TOPICS[dataset], file_name]
-                )
-            else:
-                print(f"Skipping evaluation as {dataset} is not in TOPICS.")
         if num_passes > 1:
             requests = [
-                Request(copy.deepycopy(r.query), copy.deepcopy(r.candidates))
+                Request(copy.deepcopy(r.query), copy.deepcopy(r.candidates))
                 for r in rerank_results
             ]
-
+    print(f"Reranking with {num_passes} passes complete!")
+    rerank_results[0].candidates = rerank_results[0].candidates[:top_k_rerank]

```

Comment:
Please truncate the rerank results for all indexes rather than the first one only